bin/hdfcoinc/pycbc_fit_sngls_split_binned

```diff
@@ -0,0 +1,354 @@
+parser.add_argument('--split-two-nbins', default=4, type=int,
+                    help="number of bins over which to split split-param-two. "
+                         "Default: 4")
+parser.add_argument('--split-one-log', action='store_true', default=False,
+                    help="is split-param-one split in log-space?. Default: False")
+parser.add_argument('--split-two-log', action='store_true', default=False,
+                    help="is split-param-two split in log-space?. Default: False")
+parser.add_argument("--sngl-stat", default="new_snr",
+                    choices=sngl_statistic_dict.keys(),
+                    help="Function of SNR and chisq to perform fits with")
+parser.add_argument('--ifo', type=str, help="Interferometer")
+parser.add_argument('--plot-max-x',default=1000,
+                    help="Maximum stat value for use in plots, if this "
+                         "is larger than 1.05 times the largest stat value, "
+                         "then that will be used instead. "
+                         "Default: a large value")
+parser.add_argument("--veto-file", type=str,
+                    help="File(s) in .xml format with veto segments to apply "
+                    "to triggers before fitting")
+parser.add_argument("--veto-segment-name", type=str,
+                    help="Name(s) of veto segments to apply. Optional, if not "
+                    "given all segments for a given ifo will be used")
+parser.add_argument("--stat-fit-threshold", type=float, required=True,
+                    help="Only fit triggers with statistic value above this "
+                    "threshold. Required")
+parser.add_argument("--fit-function",
+                    choices=["exponential", "rayleigh", "power"],
+                    help="Functional form for the maximum likelihood fit")
+args = parser.parse_args()
+
+args.approximant = None
+
+# set up logging
+if args.verbose:
+    log_level = logging.DEBUG
+else:
+    log_level = logging.WARN
+logging.basicConfig(format='%(asctime)s : %(message)s', level=log_level)
+
+# Open files
+logging.info('Opening trigger file: %s' % args.trigger_file)
+trigf = h5py.File(args.trigger_file, 'r')
+
+logging.info('Opening template file: %s' % args.bank_file)
+full_bank = h5py.File(args.bank_file, 'r')
+
+# store all possible parameters in a single dict for later use
+logging.info('setting up template bank parameters from the parameter file')
+params = {
+        "mass1"  : full_bank['mass1'][:],
+        "mass2"  : full_bank['mass2'][:],
+        "spin1z" : full_bank['spin1z'][:],
+        "spin2z" : full_bank['spin2z'][:]
+}
+full_bank.close()
+
+# only calculate params that need calculation if they are actually used
+usedparams = [args.split_param_two, args.split_param_one, args.bin_param]
+extparams = [par for par in ['mchirp', 'eta', 'chi_eff', 'template_duration'] if par in usedparams]
+
+logging.info('calculating extra parameters {} as required'.format(extparams))
```

Comment:
think this is not needed, given that every ex_p will be logged in the following loop.